File: src/main/java/raft/war/binary/parser/w3g/parser/replay/Unknown1cRecord.java
package raft.war.binary.parser.w3g.parser.replay;

import raft.war.binary.parser.w3g.parser.packed.IRecord;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import java.util.Objects;

public class Unknown1cRecord implements IRecord {

    public static final int TYPE = 0x1C;

    private int unknown;

    @Override
    public int getRecordId() {
        return TYPE;
    }

    @Override
    public void parse(ByteBuffer inBuffer) {
        unknown = inBuffer.getInt();
    }

    public ByteBuffer assembly(ByteBuffer outBuffer) {
        if (outBuffer == null)
            outBuffer = ByteBuffer.allocate(4).order(ByteOrder.LITTLE_ENDIAN);

        outBuffer.putInt(unknown);

        return outBuffer;
    }

    public int getUnknown() {
        return unknown;
    }

    public Unknown1cRecord setUnknown(int unknown) {
        this.unknown = unknown;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        Unknown1cRecord that = (Unknown1cRecord) o;
        return unknown == that.unknown;
    }

    @Override
    public int hashCode() {
        return Objects.hash(unknown);
    }
}

File: src/main/kotlin/raft/war/binary/parser/w3g/record/Unknown1cRecord.kt
package raft.war.binary.parser.w3g.record

import java.nio.ByteBuffer
import java.util.*

class Unknown1cRecord : RecordBase {
    override var timestamp: Long = 0

    var unknown: Int = 0

    override fun parse(inBuffer: ByteBuffer) {
        unknown = inBuffer.getInt()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as Unknown1cRecord
        return unknown == that.unknown
    }

    override fun hashCode(): Int {
        return Objects.hash(unknown)
    }

    companion object {
        const val ID: Int = 0x1C
    }

    override fun toString(): String = "Time: $timestamp,\tUnknown1cRecord\n"
}
